src/components/widget/RawWidget.js

```diff
@@ -255,7 +255,10 @@ export class RawWidget extends Component {
       initialFocus,
     } = this.props;
 
-    let widgetValue = data != null ? widgetData[0].value : null;
+    let widgetValue =
+      widgetData[0].value != null && widgetType !== 'Quantity'
+        ? widgetData[0].value
+        : data;
```

Comment:
@oliviupis  @ionut-movila  @siemiatj  
wondering if it would make sense to document what's in
* widgetData
* widgetData[0]
* widgetData[0].value
* data

More, when shall we use `data` and when `widgetData[0].value`? Why the widgetType is relevant?
I feel like this code part is quite fragile and without documentation makes it even harder for a "new" developer to do something valuable.

wdyt?

Reply:
Yes we will have to discuss with Kuba what is the reason for this logic. Basically in some cases we show the value after saving (from the API call, that in some cases for a number it adds the decimal part) and in other cases we show the local value received as a prop?

@siemiatj Good point to know how would be the best here? Sync for this when you back from US?

@teosarca I agree right now it looks a bit fragile more like a workaround, that we should look over soon.

Reply:
Ha, I wish I had answers to these questions. That's how this was built :)